src/components/accordion/accordion.js

```diff
@@ -0,0 +1,241 @@
+    return
+  }
+
+  this.moduleId = $module.getAttribute('id')
+
+  nodeListForEach(this.$sections, function ($section, i) {
+    // Set header attributes
+    var header = $section.querySelector('.govuk-accordion__section-header')
+    this.setHeaderAttributes(header, i)
+
+    this.setExpanded(this.isExpanded($section), $section)
+
+    // Handle events
+    header.addEventListener('click', this.onToggleExpanded.bind(this, $section))
+  }.bind(this))
+
+  // Create "Open all" button and set attributes
+  this.$openAllButton = document.createElement('button')
+  this.setOpenAllButtonAttributes(this.$openAllButton)
+
+  // Create controls and set attributes
+  var accordionControls = document.createElement('div')
+  accordionControls.setAttribute('class', 'govuk-accordion__controls')
+  accordionControls.appendChild(this.$openAllButton)
+  this.$module.insertBefore(accordionControls, this.$module.firstChild)
+
+  this.$module.classList.add('with-js')
+
+  // Handle events
+  this.$openAllButton.addEventListener('click', this.openOrCloseAllSections.bind(this))
+
+  // See if there is any state stored in sessionStorage and set the sections to
+  // open or closed.
+  this.readState()
+
+  // See if OpenAll button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+// Open/close section
+Accordion.prototype.onToggleExpanded = function ($section) {
+  var expanded = this.isExpanded($section)
+  this.setExpanded(!expanded, $section)
+
+  // Store the state in sessionStorage when a change is triggered
+  this.storeState()
+
+  // See if OpenAll button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+
+// Toggle aria-expanded when section opened/closed
+Accordion.prototype.setExpanded = function (expanded, $section) {
+  var $button = $section.querySelector('.govuk-accordion__section-header-button')
+  $button.setAttribute('aria-expanded', expanded)
+
+  if (expanded) {
+    $section.classList.add('govuk-accordion__section--expanded')
+  } else {
+    $section.classList.remove('govuk-accordion__section--expanded')
+  }
+
+  // This is set to trigger reflow for IE8, which doesn't
+  // always reflow after a setAttribute call.
+  this.$module.className = this.$module.className
+}
+
+Accordion.prototype.isExpanded = function ($section) {
+  return ($section.classList.contains('govuk-accordion__section--expanded'))
+}
+
+Accordion.prototype.setHeaderAttributes = function ($header, index) {
+  var $button = $header.querySelector('.govuk-accordion__section-header-button')
+
+  // Copy existing div element to an actual button element, for improved accessibility.
+  // TODO: this probably needs to be more robust, and copy all attributes and child nodes?
+  var $buttonAsButton = document.createElement('button')
+  $buttonAsButton.setAttribute('class', $button.getAttribute('class'))
```

Comment:
this needs the `type` attribute to be set to 'button' otherwise it could submit forms

Reply:
@nickcolley good spot. Dang, I thought that had been done already.

Reply:
@frankieroberto Thanks for adding this to the "Open all" button. This should be done for the accordion button as well.